frontend/views/containers/chatroom/MessagePoll.vue

```diff
@@ -82,6 +99,58 @@ export default ({
     },
     addEmoticon (emoticon) {
       this.$emit('add-emoticon', emoticon)
+    },
+    switchOnChangeMode () {
+      this.ephemeral.isChangeMode = true
+    },
+    switchOffChangeMode () {
+      this.ephemeral.isChangeMode = false
+    },
+    checkAndSetupPollExpirationTimer () {
+      const markPollClosed = () => {
+        sbp('gi.actions/chatroom/closePoll', {
+          contractID: this.currentChatRoomId,
+          data: { hash: this.messageHash }
+        })
+      }
+      const checkAndSetTimer = () => {
+        if (Date.now() - this.pollData.expires_date_ms) {
+          markPollClosed()
+        } else {
+          this.expirationTimeoutId = setTimeout(checkAndSetTimer, MINS_MILLIS)
+        }
+      }
+      const timeDiff = Date.now() - this.pollData.expires_date_ms
+
+      if (timeDiff >= 0) {
+        // if the poll has been expired, mark it 'closed' immediately.
+        markPollClosed()
+      } else if (Math.abs(timeDiff) < DAYS_MILLIS) {
```

Comment:
I randomly defined the poll **'expiring soon'** in below comment as **'within 24hrs'** here. pls let me know if this should be other value. will make a tweak.

Reply:
Yeah. The solution using `mainNotificationMixin.js` would be nice. But the problem is that we don't store any messages (including polls data) in contract state and we don't have any `getters` either.

So I think it's also fine to close the poll here in this part.